Storage/ThirdParty/SwiftData.swift

```diff
@@ -157,22 +189,28 @@ open class SwiftData {
             deferred.running = true
             defer {
                 deferred.running = false
+                deferred.connection = nil
             }
 
-            if !self.closed && self.sharedConnection == nil {
-                self.sharedConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readWriteCreate.toSQL(), key: self.key, prevKey: self.prevKey, schema: self.schema, files: self.files)
+            if !self.closed {
+                if useSecondaryConnection && self.secondaryConnection == nil {
+                    self.secondaryConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readOnly, key: self.key, prevKey: self.prevKey, schema: self.schema, files: self.files)
+                    deferred.connection = self.secondaryConnection
+                } else if self.primaryConnection == nil {
+                    self.primaryConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readWriteCreate, key: self.key, prevKey: self.prevKey, schema: self.schema, files: self.files)
+                    deferred.connection = self.primaryConnection
+                }
             }
 
-            guard let connection = SwiftData.ReuseConnections ? self.sharedConnection :
-                ConcreteSQLiteDBConnection(filename: self.filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey, schema: self.schema, files: self.files) else {
-                    do {
-                        _ = try callback(FailedSQLiteDBConnection())
+            guard let connection = useSecondaryConnection ? self.secondaryConnection : self.primaryConnection else {
+                do {
+                    _ = try callback(FailedSQLiteDBConnection())
 
-                        deferred.fill(Maybe(failure: NSError(domain: "mozilla", code: 0, userInfo: [NSLocalizedDescriptionKey: "Could not create a connection"])))
-                    } catch let err as NSError {
-                        deferred.fill(Maybe(failure: DatabaseError(err: err)))
-                    }
-                    return
+                    deferred.fill(Maybe(failure: NSError(domain: "mozilla", code: 0, userInfo: [NSLocalizedDescriptionKey: "Could not create a connection"])))
+                } catch let err as NSError {
+                    deferred.fill(Maybe(failure: DatabaseError(err: err)))
+                }
+                return
             }
 
```

Comment:
teeny nit: consider`deferred.connection = connection` here (I expected it here, and had to look back for it in the earlier 2 spots)